server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -463,6 +467,115 @@ describe("Routerlicious", () => {
 						]);
 						await Promise.all(promises);
 					});
+
+					it("Verify v2 signals", async () => {
+						let userSignalCounts = [0, 0, 0];
+
+						const createSignalPromise = (
+							userIndex: number,
+							socket: LocalWebSocket,
+						): Promise<ISignalMessage> => {
+							return new Promise((resolve) => {
+								socket.on("signal", (signal: ISignalMessage) => {
+									userSignalCounts[userIndex]++;
+									resolve(signal);
+								});
+							});
+						};
+						const firstSocket = webSocketServer.createConnection();
+
+						const firstConnectMessage = await connectToServer(
+							testId,
+							testTenantId,
+							testSecret,
+							firstSocket,
+							true,
+						);
+
+						const secondSocket = webSocketServer.createConnection();
+
+						const secondConnectMessage = await connectToServer(
+							testId,
+							testTenantId,
+							testSecret,
+							secondSocket,
+							true,
+						);
+
+						const thirdSocket = webSocketServer.createConnection();
+
+						await connectToServer(testId, testTenantId, testSecret, thirdSocket, true);
+
+						const user1SignalPromise = createSignalPromise(0, firstSocket);
+						const user2SignalPromise = createSignalPromise(1, secondSocket);
+						const user3SignalPromise = createSignalPromise(2, thirdSocket);
+
+						const targetedSignal: ISentSignalMessage = {
+							targetClientId: firstConnectMessage.clientId,
+							content: "TestSignal",
+						};
+
+						secondSocket.send("submitSignal", secondConnectMessage.clientId, [
+							targetedSignal,
+						]);
+						let user1Signal = await user1SignalPromise;
+						assert.equal(userSignalCounts[0], 1),
+							"User 1 should have received 1 signal";
```

Comment:
If we send a quick succession of targeted signals (targeting single client, or range of clients), what is expected perf/reliability expectation? Can we test that? Please use client perf expectations for some initial guideline.

Reply:
This is tested in the 'signal usage' tests underneath "Usage Counting".